example/main.go

```diff
@@ -68,8 +68,16 @@ func main() {
 			c := sqsconsumer.NewConsumer(s, handler)
 			c.SetLogger(log.Printf)
 
-			// start running the consumer with a context that will be cancelled when a graceful shutdown is requested
-			c.Run(fetchCtx)
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			go func() {
+				<-shutDown
+				time.AfterFunc(30*time.Second, cancel)
+				c.ShutDown()
+			}()
+
+			c.Run(ctx)
```

Comment:
This strikes me as some complicated shutdown logic. You could pass `shutDown` to `Run()` and make it more analogous to the [pipeline cancellation pattern](https://blog.golang.org/pipelines), or you cancel the context without a 30 second grace period. It might just be my opinion, but I tend to think of "graceful shutdown" as meaning you can flush buffers and close open resource handles but you still have to stop immediately, with the opposite of a "graceful shutdown" being immediate process termination without cleanup.

Reply:
Perhaps I am overloading the term "graceful shutdown".  Pending a better term, I see two definitions:
* A "system" shutdown cleans up resources
* A "business" shutdown finishes in-progress work, then performs the "system" shutdown.
  
  This is similar to the http server's [Shutdown](https://godoc.org/net/http#Server.Shutdown), where it stops accepting new requests, but lets active requests finish.

Any ideas for a better term?  Maybe drain or stop?

Reply:
Regarding the logic, I have an upcoming change to improve it.

Reply:
Gotcha. It's generally better to be consistent with the standard library, so let's stick with your solution's semantics.